code/fibonacci.cpp

```diff
@@ -0,0 +1,23 @@
+#include <iostream>
+
+/*Returns the n'th term of the Fibonacci sequence.*/
+int recursive_fibonacci(int n)
+{
+    if (n <= 1) {
+        return n;
+    } else {
+        return recursive_fibonacci(n - 1) + recursive_fibonacci(n - 2);
+    }
+}
+
+int main()
+{
+    std::cout << recursive_fibonacci(0) << "\n";
+    std::cout << recursive_fibonacci(1) << "\n";
+    std::cout << recursive_fibonacci(2) << "\n";
+    std::cout << recursive_fibonacci(3) << "\n";
+    std::cout << recursive_fibonacci(4) << "\n";
+    std::cout << recursive_fibonacci(5) << "\n";
+    std::cout << recursive_fibonacci(6) << "\n";
```

Comment:
```suggestion
    // Manual tests
    // Case 1: When n is 0 or 1
    int n = 0;
    std::cout << "Fib(" << n << ") = " << recursive_fibonacci(n) << "\n";
    
    n = 1;
    std::cout << "Fib(" << n << ") = " << recursive_fibonacci(n) << "\n";


    // Case 2: When n is greater than 1.
    n = 5;
    std::cout << "Fib(" << n << ") = " << recursive_fibonacci(n) << "\n";
 

    // Case 3: When n is negative.
    n = -3;
    std::cout << "Fib(" << n << ") = " << recursive_fibonacci(n) << "\n";


    // Case 4: When n is a decimal number.
    n = 7.6;
    std::cout << "Fib(" << n << ") = " << recursive_fibonacci(n) << "\n";
```

In order to make the code file consistent with the description, I have added all 4 cases that were being discussed. Although, we can also put the `cout`  statement in a function for modularity but I have not done that as it is a demonstration example for testing code manually without any framework.

Reply:
But `n` is an integer, so if it is done this way, the code won't even compile, complaining about line 35, right?

Reply:
No, the code compiles. I tested on my system. By default, when we use `n=7.6`, n stores the value 7 as n is an integer. There is no rounding off here. Hence, the code runs for n=7.

Reply:
The compiler truncates 7.6 to 7 during the assignment `n=7.6` since it knows `n` must be an int. But then line 36 isn't demonstrating the test we want it to, we should have 
```
    std::cout << "Fib(" << 7.6 << ") = " << recursive_fibonacci(7.6) << "\n";
```

Reply:
@lkr-ind But you've already defined `n` as an `int`, so that's why the compiler is converting it. `n` is not a float here. I'd hope your compiler is at least giving you a warning about this! (And our example code shouldn't produce compiler warnings or errors.)

I think this shows why it doesn't make sense to talk about running a function with the "wrong type" of argument in C++, because it's literally impossible. Yes, the compiler will implicitly convert between types in the right circumstances (like here), but that's not the same thing. `recursive_fibonacci()` *can't* accept anything other than an `int`, so it isn't something you can test or need to test.

Reply:
I updated the text of Case 4 to:

> Luckily, this is not possible in C++, being a strongly typed language, so one less thing to test. However, it can be an issue in other programming languages like Python, in which case some sort of [defensive programming](https://en.m.wikipedia.org/wiki/Defensive_programming) should be put in place to handle these cases.

But if it is going to cause confusion, we remove that case altogether.

Honestly, I don't know how C++ has been around for so long with behaviours like this one, which seem to be designed on purpose to cause problems...